overridable/bool_or_string.go

```diff
@@ -0,0 +1,82 @@
+package overridable
+
+import (
+	"encoding/json"
+)
+
+// BoolOrString is a set of rules that either evaluate to a string or a bool.
+type BoolOrString struct {
+	rules rules
+}
+
+// FromBoolOrString creates a BoolOrString representing a static, scalar value.
+func FromBoolOrString(v interface{}) BoolOrString {
```

Comment:
Do we want to validate that `v` is `bool` or `string` before handing it to `SimpleRule`? (Or, alternatively, we could expose per-type functions, although we'd have a potential name collision with `Bool`, so it would be something horrible like `BoolOrStringFromBool` and `BoolOrStringFromString`.)

Reply:
I see what you mean. I figured it's alright since it's already covered in the schema validation, but maybe another precaution isn't too bad. I'll tinker around a bit

Reply:
I think this is fine for now, given that the API would become clunkier then, when this would return an error as well (cannot use easily in a test). If you do not agree, let me know and I'll find a way.

Reply:
Merging for now, happy to address feedback post-merge :)

Reply:
I think this was the right decision, based on your comments. Thanks for taking the time to investigate!